go-controller/pkg/ovn/pods.go

```diff
@@ -414,6 +430,22 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			pod.Namespace, pod.Name, time.Since(start), libovsdbExecuteTime, podAnnoTime)
 	}()
 
+	// it is possible to try to add a pod here that has no node. For example if a pod was deleted with
+	// a finalizer, and then the node was removed. In this case the pod will still exist in a running state.
+	// Terminating pods should still have network connectivity for pre-stop hooks or termination grace period
+	if _, err := oc.watchFactory.GetNode(pod.Spec.NodeName); kerrors.IsNotFound(err) &&
```

Comment:
Just Confirming: The way we are deciding if the node is deleted or not is 1) we see if it exists in KAPI AND 2) if doesn't have a switch subnet in OVNK cache.
I am assuming we need to do 1 && 2 in case the node is gone from kapi but we still have its wiring in OVNK ? Like node retry deletion or something. So if there is a subnet that means we still have a switch for the node...

Reply:
I am ok with this approach, iot looks to me as the reasonable effort to do now.

Overall going forward I would try not to depend on trying to ascertain if the node exists or not here. That's subject to race now or in the future and is a node handler which should be aware of it and act in consequence. What i would probably do here is just blacklist the IP in the allocator temporarily until we are sure what is going on with it at a later time.

Reply:
I guess that all that we can assume is that we are not aware of the node state, and we try to allocate the IPs if at all possible assuming that if the node were to be deleted the IPs would still be de-allocated.